extensions/blocks/podcast-player/view.js

```diff
@@ -58,26 +59,224 @@ if ( window.jetpackPodcastPlayers !== undefined ) {
 	push: initializeBlock,
-	playerInstances,
 };
 
-// Add global handler for clicks.
-window.addEventListener( 'click', function( e ) {
-	// Prevent handling clicks if a modifier is in use.
-	if ( e.shiftKey || e.metaKey || e.altKey ) {
+function createSVGs() {
+	// for some reason document.createElement( 'svg' ) would not work. The HTML would work correctly, but the SVG would not render
+	const svgTemplate = document.createElement( 'div' );
+	svgTemplate.setAttribute( 'style', 'position: absolute; width: 0; height: 0; overflow: hidden;' );
+
+	const soundIcon =
+		'<symbol id="podcast-player-icon__sound" viewBox="0 0 24 24"><path d="M0 0h24v24H0V0z" fill="none"/><path d="M3 9v6h4l5 5V4L7 9H3zm7-.17v6.34L7.83 13H5v-2h2.83L10 8.83zM16.5 12c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77 0-4.28-2.99-7.86-7-8.77z"/></symbol>';
+	const errorIcon =
+		'<symbol id="podcast-player-icon__error" viewBox="0 0 24 24"><path d="M0 0h24v24H0V0z" fill="none"/><path d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"/></symbol>';
+	svgTemplate.innerHTML = `<svg style="position: absolute; width: 0; height: 0; overflow: hidden;" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs>${ soundIcon }${ errorIcon }</defs></svg>`;
+	// put it in the body
+	document.body.appendChild( svgTemplate );
+}
+
+createSVGs();
+
+function handleEpisodeLinkKeydown( e ) {
+	// early return as quickly as possible to prevent potential performance issues.
+	// we only care about the spacebar
+	if ( event.key !== ' ' || ! e.target.classList.contains( 'podcast-player__episode-link' ) ) {
 		return;
 	}
 
-	// Check if the clicked element was episode link.
-	const audioUrl = e.target.getAttribute( 'data-jetpack-podcast-audio' );
-	if ( audioUrl ) {
-		const block = e.target.closest( '.wp-block-jetpack-podcast-player' );
-		const player = block && block.id && playerInstances[ block.id ];
-		if ( player ) {
-			player.audio.pause();
-			player.audio.src = audioUrl;
-			player.audio.play();
-			e.preventDefault();
+	e.stopPropagation();
+
+	handleEpisodeLinkEvent( e );
+}
+
+function handleEpisodeLinkEvent( e ) {
+	const episodeLinkEl = e.target.closest( '[data-jetpack-podcast-audio]' );
+
+	if ( episodeLinkEl ) {
+		// Prevent handling clicks if a modifier is in use.
+		if ( e.shiftKey || e.metaKey || e.altKey ) {
+			return;
+		}
+
+		e.preventDefault();
+		handleEpisodeLinkClick( episodeLinkEl );
+	}
+}
+
+// Add document event listeners
+document.addEventListener( 'click', handleEpisodeLinkEvent );
+document.addEventListener( 'keydown', handleEpisodeLinkKeydown );
+
+function handleEpisodeLinkClick( episodeLinkEl ) {
+	// Get clicked episode element
+	const episodeEl = episodeLinkEl.closest( '.podcast-player__episode' );
```

Comment:
> In my previous life we mandated that if a CSS class was used as a JS hook then we prefixed it with js- to reduce the likelihood it would be unintentionally changed.

It was the same for me. I'd vote for using the `js-*` prefix for resiliency even though it can produce a lot more markup sometimes.